Source/FakeItEasy/Core/MethodInfoManager.cs

```diff
@@ -170,12 +160,23 @@ private static bool TypeImplementsInterface(Type type, Type interfaceType)
 
         private struct TypeMethodInfoPair
         {
-            public MethodInfo MethodInfo;
-            public Type Type;
+            public TypeMethodInfoPair(Type type, MethodInfo methodInfo)
+                : this()
+            {
+                Type = type;
+                MethodInfo = methodInfo;
+            }
+
+            public MethodInfo MethodInfo { get; private set; }
+            
+            public Type Type { get; private set; }
 
             public override int GetHashCode()
             {
-                return this.Type.GetHashCode() ^ this.MethodInfo.GetHashCode();
+                unchecked
+                {
+                    return (Type.GetHashCode() * 23) + MethodInfo.GetHashCode();
+                }
             }

```

Comment:
Technically the changes to TypeMethodInfoPair weren't needed, but the original GetHashCode 

1. exhibited the suboptimal "just xor things together" pattern, and
2. built the hashcode from non-readonly members, so it was technically possible to mutate the object and mess up the hashing